app/src/main/java/org/sil/hearthis/MainActivity.java

```diff
@@ -1,76 +1,220 @@
+	{
+		mainMenu = menu;
+		MenuInflater inflater = getMenuInflater();
+		inflater.inflate(R.menu.menu_main, menu);
+		return super.onCreateOptionsMenu(menu);
+	}
+
+	public void setProject(Project project)
+	{
+		ArrayList<BookInfo> info = project.getBooks();
+		LayoutInflater inflater = (LayoutInflater)getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+
+		GridView oldTGrid = (GridView)findViewById(R.id.oldTGrid);
+		BookButtonAdapter oldTAdapter = new BookButtonAdapter(this, info, 0, 38);
+		oldTGrid.setAdapter(oldTAdapter);
+		oldTAdapter.notifyDataSetChanged();
+		oldTGrid.setOnItemClickListener(bookButtonListener);
+
+		GridView newTGrid = (GridView)findViewById(R.id.newTGrid);
+		BookButtonAdapter newTAdapter = new BookButtonAdapter(this, info, 39, 65);
+		newTGrid.setAdapter(newTAdapter);
+		newTAdapter.notifyDataSetChanged();
+		newTGrid.setOnItemClickListener(bookButtonListener);
+	}
+
+	public AdapterView.OnItemClickListener bookButtonListener = new AdapterView.OnItemClickListener()
+	{
 		@Override
-		public void onClick(View v) {
+		public void onItemClick(AdapterView<?> parent, View v, int position, long id)
+		{
 			BookInfo book = (BookInfo)v.getTag();
-			Intent chooseChapter = new Intent(MainActivity.this, ChooseChapterActivity.class);
-			chooseChapter.putExtra("bookInfo", book);
+			Intent chooseChapter = new Intent(MainActivity.this, ChaptersActivity.class);
+			chooseChapter.putExtra("book", book);
 			startActivity(chooseChapter);
 		}
 	};
+
+	public class BookButtonAdapter extends BaseAdapter
+	{
+		private Context context;
+		// references to button features
+		private BookInfo[] buttonInfo;
+
+		public BookButtonAdapter(Context c, ArrayList<BookInfo> infoList, int first, int last)
+		{
+			context = c;
+			buttonInfo = new BookInfo[last - first + 1];
+			for (int j = 0; j <= last - first; j++)
+				buttonInfo[j] = infoList.get(first + j);
+		}
+
+		public int getCount()
+		{
+			return buttonInfo.length;
+		}
+
+		public Object getItem(int position)
+		{
+			return buttonInfo[position];
+		}
+
+		public long getItemId(int position)
+		{
+			return 0;
+		}
+
+		public View getView(int position, View convertView, ViewGroup parent)
+		{
+			TextView button;
+			BookInfo myInfo = buttonInfo[position];
+			if (convertView == null)
+			{
+				button = new TextView(context);
+				button.setPadding(8, 8, 8, 8);
+				button.setBackgroundColor(setButtonColour(myInfo.getBookNumber()));
+				button.setTextColor(getResources().getColor(R.color.navButtonTextColor));//0xffffffff);
+				button.setHeight(80);
+				button.setTextSize(20);
+				button.setGravity(Gravity.CENTER);
+			}
+			else
+			{
+				button = (TextView)convertView;
+			}
+			button.setTag(myInfo);
+			button.setText(myInfo.getAbbr() != null ?
+					myInfo.getAbbr() :
+					myInfo.getName().substring(0, 3));
+			return button;
+		}
+
+		private int setButtonColour(int bookNumber)
+		{
+			int navbuttoncolour = R.color.navButtonColor;
+			if (bookNumber < 5)
+			{
```

Comment:
Assuming we agree on the Google coding standard, I'd naturally like you to use it for your additions, too.